File: CSharpFundamentals/CSharpAdvanced/StacksAndQueues_Lab/SimpleCalculator.cs
using System;
using System.Collections.Generic;
using System.Linq;

class SimpleCalculator
{
    static void Main()
    {
        string[] input = Console.ReadLine().Split();
        Stack<string> stack = new Stack<string>();

        for (int i = 0; i < input.Length; i++)
        {
            try
            {
                int num = int.Parse(input[i]);
                stack.Push(num)
            }
            catch (Exception)
            {

                throw;
            }
        }

    }
}

File: JavaFundamentals/JavaAdvanced/Stacks-and-queues-lab/src/SimpleCalculator.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Collections;

public class SimpleCalculator {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        String[] input = reader.readLine().split("\\s+");

        ArrayDeque<String> tockens = new ArrayDeque<>();
        Collections.addAll(tockens, input);

        while (tockens.size() > 1){
            int first = Integer.parseInt(tockens.pop());
            String operator = tockens.pop();
            int second = Integer.parseInt(tockens.pop());

            int result;

            if (operator.equals("+")){
                result = first + second;
            } else {
                result = first - second;
            }

            tockens.push(String.valueOf(result));
        }

        int sum = Integer.parseInt(tockens.pop());
        System.out.println(sum);
    }
}
